sormas-backend/src/main/java/de/symeda/sormas/backend/sample/SampleService.java

```diff
@@ -137,6 +158,287 @@ public SampleService() {
+				final SampleJurisdictionFlagsDto sampleJurisdictionFlagsDto = s.getSampleJurisdictionFlagsDto();
+				if (s.getAssociatedCase() != null) {
+					emptyValuePseudonymizer
+						.pseudonymizeDto(CaseReferenceDto.class, s.getAssociatedCase(), sampleJurisdictionFlagsDto.getCaseInJurisdiction(), null);
+				}
+
+				ContactReferenceDto associatedContact = s.getAssociatedContact();
+				if (associatedContact != null) {
+					emptyValuePseudonymizer.pseudonymizeDto(
+						ContactReferenceDto.PersonName.class,
+						associatedContact.getContactName(),
+						sampleJurisdictionFlagsDto.getContactInJurisdiction(),
+						null);
+
+					if (associatedContact.getCaseName() != null) {
+						pseudonymizer.pseudonymizeDto(
+							ContactReferenceDto.PersonName.class,
+							associatedContact.getCaseName(),
+							sampleJurisdictionFlagsDto.getContactCaseInJurisdiction(),
+							null);
+					}
+				}
+
+				if (s.getAssociatedEventParticipant() != null) {
+					emptyValuePseudonymizer.pseudonymizeDto(
+						EventParticipantReferenceDto.class,
+						s.getAssociatedEventParticipant(),
+						sampleJurisdictionFlagsDto.getEvenParticipantInJurisdiction(),
+						null);
+				}
+			}, true);
+
+		return samples;
+	}
+
+	public List<SampleListEntryDto> getEntriesList(SampleCriteria sampleCriteria, Integer first, Integer max) {
+		if (sampleCriteria == null) {
+			return Collections.emptyList();
+		}
+
+		final CriteriaBuilder cb = em.getCriteriaBuilder();
+		final CriteriaQuery<Object[]> cq = cb.createQuery(Object[].class);
+		final Root<Sample> sample = cq.from(Sample.class);
+
+		SampleQueryContext sampleQueryContext = new SampleQueryContext(cb, cq, sample);
+		SampleJoins<Sample> joins = (SampleJoins<Sample>) sampleQueryContext.getJoins();
+
+		cq.distinct(true);
+
+		List<Selection<?>> selections = new ArrayList<>(
+			Arrays.asList(
+				sample.get(Sample.UUID),
+				sample.get(Sample.SAMPLE_MATERIAL),
+				sample.get(Sample.PATHOGEN_TEST_RESULT),
+				sample.get(Sample.SPECIMEN_CONDITION),
+				sample.get(Sample.SAMPLE_PURPOSE),
+				joins.getReferredSample().get(Sample.UUID),
+				sample.get(Sample.RECEIVED),
+				sample.get(Sample.RECEIVED_DATE),
+				sample.get(Sample.SHIPPED),
+				sample.get(Sample.SHIPMENT_DATE),
+				sample.get(Sample.SAMPLE_DATE_TIME),
+				joins.getLab().get(Facility.NAME),
+				joins.getLab().get(Facility.UUID),
+				sample.get(Sample.SAMPLING_REASON),
+				sample.get(Sample.SAMPLING_REASON_DETAILS),
+				sample.get(Sample.ADDITIONAL_TESTING_REQUESTED),
+				cb.isNotEmpty(sample.get(Sample.ADDITIONAL_TESTS))));
+
+		// Tests count subquery
+		Subquery<Long> testCountSq = cq.subquery(Long.class);
+		Root<PathogenTest> testCountRoot = testCountSq.from(PathogenTest.class);
+		testCountSq.where(
+			cb.equal(testCountRoot.join(PathogenTest.SAMPLE, JoinType.LEFT).get(Sample.ID), sample.get(Sample.ID)),
```

Comment:
Can you test whether you can also write this line like this:

`cb.equal(testCountRoot.get(PathogenTest.SAMPLE), sample)`

Not sure whether that works, but if it does, it would spare us additional joins.

Reply:
Seems to be working, I have made the change for sample directory as well.